pkg/agent/init-container.go

```diff
@@ -31,13 +86,21 @@ func (a *Agent) ContainerInitSidecar() (corev1.Container, error) {
 		ReadOnly:  false, // Changed to false to allow writing
 	})
 
-	parsedAgentConfig := util.BuildAgentConfigFromConfigMap(a.configMap, a.injectMode)
+	parsedAgentConfig, err := util.BuildAgentConfigFromConfigMap(a.configMap, a.injectMode)
+	if err != nil {
+		return corev1.Container{}, fmt.Errorf("failed to build agent config: %w", err)
+	}
 
 	agentConfigYaml, err := yaml.Marshal(parsedAgentConfig)
 	if err != nil {
 		return corev1.Container{}, fmt.Errorf("failed to marshal yaml agent config: %w", err)
 	}
 
+	filePathCreationScript, err := getFilePathCreationScript(*a.configMap)
+	if err != nil {
+		return corev1.Container{}, fmt.Errorf("failed to get file path creation script: %w", err)
+	}
+
 	script := []string{
 		"#!/bin/sh",
 		"set -ex", // -x to trace command execution
```

Comment:
with this set and echo commands, the ldap password will be in the shell, right? also instead of echo we should we heredoc for writing the files https://linuxize.com/post/bash-heredoc/